### ISSUE TITLE
Mirroring support

### ISSUE THREAD

**Issue by XVilka** [ocaml/opam#4103]
Currently it is impossible to make a local or regional mirror for both opam repository and source files in a batch, like some of the Linux distributions or package managers. It would be amazing and useful to setup a couple mirrors on a various regional CDNs, like some other languages/distributions did, And add the instruction on how to add them into the global mirror lists:

-    https://www.gentoo.org/downloads/mirrors/
-    https://www.debian.org/mirror/list
-    https://npmjs.cf/

See example pages on how to setup the mirror:

-    https://www.debian.org/mirror/ftpmirror
-    https://wiki.gentoo.org/wiki/Project:Infrastructure/Mirrors/Source

See the example of a big mirror from Tencent with various languages, distributions and frameworks: https://mirrors.tencent.com/

I [asked this on Discourse](https://discuss.ocaml.org/t/how-to-setup-local-opam-mirror/4423) but got no easy solution

**Comment by rjbou** [comment]
The steps are dispersed in the discourse thread. To summarize:
* Set up the mirror
  * copy of opam-repository (git clone)
  * `opam admin cache` if you want to cache archives
  * `opam admin index` for http
  * have a webserver
  * _for opam.ocaml.org, scripts are in this [repo](https://github.com/ocaml/opam.ocaml.org/)_
* Use the mirror
  * `opam repo add mirror <url>`, to have it as a fallback, or with `--rank 1`
  * with `--set-default` to have it as the default one for switch creation

It can be the occasion to have a [blog post](https://github.com/ocaml/platform-blog/) for that.

**Comment by XVilka** [comment]
Yes, I would be awesome to have a good step-by-step guide, especially since it will be the IT team who setups this kind of mirror, not seasoned OCaml users/developers.

**Comment by dbuenzli** [comment]
> It can be the occasion to have a [blog post](https://github.com/ocaml/platform-blog/) for that.

Blog posts are poor substitutes for manuals -- for one thing they are difficult to update. It's better to have that in the manual.

**Comment by XVilka** [comment]
Let me know if I can help somehow.
